Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp

```diff
@@ -103,6 +115,16 @@ namespace AZ
             AzFramework::WindowNotificationBus::Handler::BusDisconnect(m_windowHandle);
         }
 
+        void WindowContext::OnVsyncIntervalChanged(uint32_t interval)
+        {
+            if (m_swapChain->GetDescriptor().m_verticalSyncInterval != interval)
+            {
+                m_swapChain->SetVerticalSyncInterval(interval);
+
+                m_swapChain->Resize(m_swapChain->GetDescriptor().m_dimensions);
```

Comment:
Why is this resizing with the same size? Is this to force swapchain recreation? If that's the case each platform should handle the change of vertical sync internally (some platforms don't even support it).

Reply:
Yea this forces swapchain recreation. Which platforms do we support that don't handle vblanks?

Reply:
The only place where the m_verticalSyncInterval is used is in o3de\Gems\Atom\RHI\DX12\Code\Source\Platform\Windows\RHI\SwapChain_Windows.cpp file's 
uint32_t SwapChain::PresentInternal() function. 
HRESULT hresult = m_swapChain->Present(GetDescriptor().m_verticalSyncInterval, 0);

which means we don't need to re-create swapchain to apply the change.

Reply:
Yup thanks I had missed that. I should have tried first in retrospect.